packages/typespec-ts/test/modularUnit/scenarios/operations/queryParam/queryParamCollectionFormat.md

```diff
@@ -0,0 +1,221 @@
+} from "@azure-rest/core-client";
+
+export function _createSend(
+  context: Client,
+  ssvArray: number[],
+  pipeArray: string[],
+  options: CreateOptionalParams = { requestOptions: {} },
+): StreamableMethod {
+  const path = expandUrlTemplate(
+    "/query{?ssvArray,ssvOptionalArray,pipeArray}",
+    {
+      ssvArray: buildSsvCollection(
+        ssvArray.map((p: any) => {
+          return p;
+        }),
+      ),
+      ssvOptionalArray: !options?.ssvOptionalArray
+        ? options?.ssvOptionalArray
+        : buildSsvCollection(
+            options?.ssvOptionalArray.map((p: any) => {
+              return p;
+            }),
+          ),
+      pipeArray: buildPipeCollection(
+        pipeArray.map((p: any) => {
+          return p;
+        }),
+      ),
+    },
+    {
+      allowReserved: options?.requestOptions?.skipUrlEncoding,
+    },
+  );
+  return context.path(path).post({ ...operationOptionsToRequestParameters(options) });
+}
+
+export async function _createDeserialize(result: PathUncheckedResponse): Promise<void> {
+  const expectedStatuses = ["204"];
+  if (!expectedStatuses.includes(result.status)) {
+    throw createRestError(result);
+  }
+
+  return;
+}
+
+export async function create(
+  context: Client,
+  ssvArray: number[],
+  pipeArray: string[],
+  options: CreateOptionalParams = { requestOptions: {} },
+): Promise<void> {
+  const result = await _createSend(context, ssvArray, pipeArray, options);
+  return _createDeserialize(result);
+}
+
+export function _readSend(
+  context: Client,
+  simpleArray: string[],
+  options: ReadOptionalParams = { requestOptions: {} },
+): StreamableMethod {
+  const path = expandUrlTemplate(
+    "/query{?simpleArray,simpleOptionalArray}",
+    {
+      simpleArray: simpleArray.map((p: any) => {
+        return p;
+      }),
+      simpleOptionalArray: !options?.simpleOptionalArray
+        ? options?.simpleOptionalArray
+        : options?.simpleOptionalArray.map((p: any) => {
+            return p;
+          }),
+    },
+    {
+      allowReserved: options?.requestOptions?.skipUrlEncoding,
+    },
+  );
+  return context.path(path).get({ ...operationOptionsToRequestParameters(options) });
+}
+
+export async function _readDeserialize(result: PathUncheckedResponse): Promise<void> {
+  const expectedStatuses = ["204"];
+  if (!expectedStatuses.includes(result.status)) {
+    throw createRestError(result);
+  }
+
+  return;
+}
+
+export async function read(
+  context: Client,
+  simpleArray: string[],
+  options: ReadOptionalParams = { requestOptions: {} },
+): Promise<void> {
+  const result = await _readSend(context, simpleArray, options);
+  return _readDeserialize(result);
+}
+```
+
+# Should explode if both explode and collection format encoding are applied to the query parameters
+
+## TypeSpec
+
+```tsp
+@route("/query")
+interface QueryOperations {
+  @get
+  read(
+    @query(#{explode: true})
+    @encode(ArrayEncoding.spaceDelimited)
+    ssvArray: int32[];
```

Comment:
I wonder if ssvArray is a string[], will we import the multi collection helper or ssv collection helper?